src/Native/jitinterface/CMakeLists.txt

```diff
@@ -18,13 +18,21 @@ if(WIN32 AND CLR_CMAKE_PLATFORM_ARCH_AMD64)
     add_definitions(-DFEATURE_REDHAWK)
 endif(WIN32 AND CLR_CMAKE_PLATFORM_ARCH_AMD64)
 
+# Convert the sources to absolute paths, the subdirs want to use them.
+convert_to_absolute_path(NATIVE_SOURCES ${NATIVE_SOURCES})
+
 add_library(jitinterface
     SHARED
     ${NATIVE_SOURCES}
 )
 
 install (TARGETS jitinterface DESTINATION tools)
 if(WIN32)
-    target_link_libraries(jitinterface ntdll.lib) 
+    target_link_libraries(jitinterface ntdll.lib)
     install (FILES ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/jitinterface.pdb DESTINATION tools)
 endif(WIN32)
+
+if (CLR_CMAKE_PLATFORM_ARCH_I386)
+  # On x86, build jitinterface for cross compiling arm/armel
+  add_subdirectory(armeljitinterface)
```

Comment:
Is this really needed? I think the regular jitinterface should work for cross compiling too. We try hard to not have platform-specific ifdefs in the JIT/EE interface layout.

Some of the JIT/EE interface flags have different meaning on x86 vs. arm, but it should not matter for the jitinterface.dll/so. It does not interpret the flags.

Reply:
hmm... I also think superfluous ifdefs are not good, but here there is no ifdefs addition to the code. Build with the jitinterface and x86 target works. But on the other hand, [here](https://github.com/dotnet/corert/blob/master/src/Native/jitinterface/JITCodeManager.cpp#L322) is a certain logic for an unwind - and I don't know, will it work properly with the x86 target? What do you think?

Reply:
JITCodeManager.cpp is part of experimental prototype for CoreRT JIT (https://github.com/dotnet/corert/commit/d4da37e00d89a168c178d7177b8f77793e2a08e9). It's not used in the compiler and we only include this file under `if(WIN32 AND CLR_CMAKE_PLATFORM_ARCH_AMD64)`, mostly to just prevent bit rot. It's not actually used.

Since JIT is a runtime component, platform specific ifdefs would be fine there.